src/OpenTelemetry.AutoInstrumentation/Instrumentation.cs

```diff
@@ -131,26 +131,43 @@ public static void Initialize()
 
             if (TracerSettings.Value.TracesEnabled)
             {
-                var builder = Sdk
-                    .CreateTracerProviderBuilder()
-                    .ConfigureResource(ResourceConfigurator.Configure)
-                    .UseEnvironmentVariables(LazyInstrumentationLoader, TracerSettings.Value, _pluginManager)
-                    .InvokePlugins(_pluginManager);
-
-                _tracerProvider = builder.Build();
-                Logger.Information("OpenTelemetry tracer initialized.");
+                if (GeneralSettings.Value.InjectSdk)
+                {
+                    var builder = Sdk
+                        .CreateTracerProviderBuilder()
+                        .ConfigureResource(ResourceConfigurator.Configure)
+                        .UseEnvironmentVariables(LazyInstrumentationLoader, TracerSettings.Value, _pluginManager)
+                        .InvokePlugins(_pluginManager);
+
+                    _tracerProvider = builder.Build();
+                    Logger.Information("OpenTelemetry tracer initialized.");
+                }
+                else
+                {
+                    AddLazilyLoadedTraceInstrumentations(LazyInstrumentationLoader, _pluginManager, TracerSettings.Value.EnabledInstrumentations);
```

Comment:
I am not sure if we should call this. More in next comment.

Reply:
I understood it to be expected behavior in this scenario when we discussed the requirements with @pjanotti

Reply:
@pjanotti can you please double-check. I find it weird.

@lachmatt I am approving as I think this can be addressed in a separate PR.

Reply:
Per discussion during SIG: Mateusz will check the behavior but barring major issues with that we go as it is and refine it later if needed. Initially the setting to setup the SDK or not should only affect that.